Workshops/2024-07-10_LLM-Evaluation/README.md

```diff
@@ -0,0 +1,64 @@
+# TMLS - LLM Evaluation
+
+In this stream we'll cover various aspects of LLM evaluation including:
+
+* Traditional evaluation, human evaluation, evaluating LLMs
+* Evaluation metrics and considerations
+* Evaluation tool and platform comparison, including open-sourced alternatives
+* Hands-on end-to-end examples
+
+
+## Environment Setup & Installation
+
+1. Setup your environment
+
+If you use Conda:
+
+```bash
+conda create -n llm-evaluation python=3.10
+conda activate llm-evaluation
+pip install -r requirements.txt
+```
+
+If you use Poetry:
+
+```bash
+poetry shell
+poetry install
+```
+
+If you use pure Python:
+
+```bash
+python -m venv env
+source env/bin/activate
+pip install -r requirements.txt
+```
+
+2. Create private environment file (this will not be committed!)
+```
+cp .env-template .env
+```
+
+Add any necessary API keys there following the given format. Specifically, you will need a key for your LLM (if you're using an API), a key for [LangSmith](https://smith.langchain.com/) (optional; it's called `LANGCHAIN_API_KEY` in the config), and a key for [Langfuse](https://langfuse.com/) (optional). Both LangSmith and Langfuse offer free API keys which can be obtained on their respective websites by creating an account. You can see the section below for mroe details.
+
+## Quick-Start:
+
+The `notebooks` directory contains a step-by-step, end-to-end example of a Q&A app with RAG. it consists of the following:
+
+1. `01-llm-app-setup.ipynb`: Set up the LLM app
+2. `02-dataset-creation.ipynb`:Create evaluation dataset
+3. `03-metrics-definition.ipynb`: Define metrics to use
+4. `04-app-evaluation.ipynb`: Evaluate the app
+
+### Required: Using an LLM
+We use langchain for all the code in the notebooks. By default we are using an OpenAI model and thus require an OpenAI key. The entire example will cost around $25 with OpenAI. However, you can change the LLM being used and the code should still work. You can refer to the [LangChain documentation](https://python.langchain.com/docs/integrations/llms/) to change the LLM being used. The first notebook explains how to swap out OpenAI LLMs for others.
+
+### Optional: Using LangSmith or Langfuse
```

Comment:
Same general question of whether we're using langsmith/langfuse here. Remove if not used.